web/static/pages/piece/piece-info.mjs

```diff
@@ -56,30 +120,360 @@ customElements.define('piece-info',class PieceInfoElement extends LitElement {
                 </tr>
                 </thead>
                 <tbody>
-                ${this.data.deals.map((item, i) => html`
+                ${this.data.deals.map((item) => html`
                     <tr>
-                <td>${item.id}</td>
-                <td>${item.boost_deal ? 'Boost' : (item.legacy_deal ? 'Legacy' : 'DDO')}</td>
-                <td>${item.miner}</td>
-                <td>${item.chain_deal_id}</td>
-                <td>${item.sector}</td>
-                <td>${item.offset}</td>
-                <td>${item.length}</td>
-                <td>${item.raw_size}</td>
-            </tr>
+                        <td><a href="/pages/mk12-deal/?id=${item.id}">${item.id}</a></td>
+                        <td>${item.boost_deal ? 'Boost' : (item.legacy_deal ? 'Legacy' : 'DDO')}</td>
+                        <td>${item.miner}</td>
+                        <td>${item.chain_deal_id}</td>
+                        <td><a href="/pages/sector/?sp=${item.miner}&id=${item.sector}">${item.sector}</a></td>
+                        <td>${item.offset}</td>
+                        <td>${this.toHumanBytes(item.length)}</td>
+                        <td>${this.toHumanBytes(item.raw_size)}</td>
+                    </tr>
                 `)}
                 </tbody>
             </table>
-            <hr>
+
+            ${this.pieceParkStates ? this.renderPieceParkStates() : ''}
+
+            ${this.mk12DealData && this.mk12DealData.length > 0 ? html`
+                <h2>Related Deals</h2>
+                ${this.mk12DealData.map((entry) => html`
+                    <h3>Deal ${entry.deal.uuid}</h3>
+                    <table class="table table-dark table-striped table-sm">
+                        <tr><th colspan="2"><h5>Top Level Info 📋</h5></th></tr>
+                        <tr><td>Created At</td><td>${formatDate(entry.deal.created_at)}</td></tr>
+                        <tr><td>UUID</td><td><a href="/pages/mk12-deal/?id=${entry.deal.uuid}">${entry.deal.uuid}</a></td></tr>
+                        <tr><td>Provider (sp_id)</td><td>f0${entry.deal.sp_id}</td></tr>
```

Comment:
Can we avoid hardcoding f0? I clean it up on main already.

Reply:
Oh yeah, meant to clean this up